### PR TITLE
MAINT: Ensure we dont download without permission

### PR DESCRIPTION
**Pull request by larsoner**
Lots of failures locally with this.

Okay for me to put `download=False` in a bunch of `data_path()` calls and add `requires_testing_data` calls where necessary?

- [x] Show failures: https://github.com/mne-tools/mne-bids/actions/runs/3152298983/jobs/5127340379
- [x] Fix failures
- [x] Revert change to use `larsoner` branch in favor of MNE-Python `main` once https://github.com/mne-tools/mne-python/pull/11213 is merged

### CODE REVIEW

**Comment by hoechenberger** [comment]
Good enough for me!

**Comment by larsoner** [comment]
A few things I found and fixed along the way:

1. Don't create `data = np.empty(...)` for `RawArray` in tests. The array can contain junk and lead to casting errors (e.g., to `float32` when wrining).
2. Don't use bare `np.random.random(...)` for the same reason. It can contain numbers that fit in `float64` but not `float32` leading to overflow errors when casting.
3. Added a `bad_frame_xfail` that we can remove once https://github.com/mne-tools/mne-python/pull/11212 is merged and any necesary changes are made here, in the meantime it allows tests here to be green
4. Added to CIs a run that executes tests *without* the testing data. This should be quick (locally takes 6 sec) and pass, and ensure that `requires_testing_data` decorators are used properly.

Pushing a commit now that hopefully will pass!

**Comment by larsoner** [comment]
... I also added dataset caching, and removed one use of `sample` data where we could just as easily use `testing`.

File: .github/workflows/unit_tests.yml [hunk 1]
```diff
@@ -158,13 +158,13 @@ jobs:
...
       run: |
-        git clone --depth 1 https://github.com/mne-tools/mne-python.git -b maint/1.0
+        git clone --depth 1 --single-branch --branch maint/1.0 https://github.com/mne-tools/mne-python.git -b maint/1.0
         python -m pip install -e ./mne-python
 
     - name: Install MNE (main)
       if: "matrix.mne-version == 'mne-main'"
       run: |
-        git clone --depth 1 https://github.com/mne-tools/mne-python.git -b main
+        git clone --depth 1 --single-branch --branch main https://github.com/mne-tools/mne-python.git -b main
```
Comment (agramfort) [inline .github/workflows/unit_tests.yml:167]: shall we keep this?

Comment (larsoner) [reply .github/workflows/unit_tests.yml:167]: Yes it's just more specific/explicit and a tiny bit faster

**Comment by larsoner** [comment]
All green, ready for review/merge from my end

File: .github/workflows/unit_tests.yml (hunk 1 above)
Comment (sappelhoff) [reply .github/workflows/unit_tests.yml:167]: but we should probably remove the trailing `-b main`, right? that's just a single flag specified twice?

**Comment by sappelhoff** [comment]
Thanks!

**Pull request merged**